pkg/controller/storagecluster/update.go

```diff
@@ -740,9 +741,9 @@ func isCSIBounceRequired(oldSpec, currentSpec *corev1.StorageClusterSpec) bool {
 		return oldSpec.CSI.Enabled != currentSpec.CSI.Enabled
 	}
 
-	// New CSI spec added to cluster. If enabled, pod will bounce
-	if oldSpec.CSI == nil && currentSpec.CSI != nil {
-		return currentSpec.CSI.Enabled
+	// Spec changed from enabled -> disabled or vice versa
+	if isCSIEnabled(oldSpec) != isCSIEnabled(currentSpec) {
```

Comment:
I think the entire function can now be simplified to just these lines, right? Wdyt?
```
func isCSIBounceRequired(..) bool {
    return isCSIEnabled(oldSpec) != isCSIEnabled(currentSpec)
}
```

Reply:
My unit tests cases fail when I switch to just this logic. Let me work through why..

Reply:
This was a test issue where the feature gate wasn't getting set back to nil. Fixed, and simplified the code as suggested.

Reply:
The feature gate is set to nil by the portworx driver, but it is mocked here. So, we will have to set it to nil in the test.